### PR TITLE
Init docker registry service and package

### PR DESCRIPTION
**Pull request by globin**
###### Motivation for this change

Re-adding the docker registry service with their new software.
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by joachifm** [comment]
Fixes https://github.com/NixOS/nixpkgs/issues/14129

**Comment by NeQuissimus** [comment]
Ha, awesome, @globin !
I happen to have "Setup up Docker registry" on my ToDos for this month :D

**Review by NeQuissimus — APPROVED** [review]

**Review by LnL7 — APPROVED** [review]
A lot of services  have a `package` option, might also be usefull for this.

**Comment by globin** [comment]
This makes more sense if there are multiple versions available, to be able to choose, I don't think this necessary here.

File: pkgs/applications/virtualization/docker-distribution/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchFromGitHub {
+    owner = "docker";
+    repo = "distribution";
+    inherit rev;
+    sha256 = "08nxcsl9bc3k9gav2mkqccm5byrlfcgy6qaqaywiyza0b0cn4kdc";
+  };
+
+  meta = with stdenv.lib; {
+    description = "The Docker toolset to pack, ship, store, and deliver content.";
```
Comment (joachifm) [inline pkgs/applications/virtualization/docker-distribution/default.nix]: Nit: trailing punctuation.

**Pull request merged**